Correction to the effective layer top and bottom calculation.

EdwardColon-NOAA opened the pull request:
Correction to the effective layer top and bottom calculation.  The key modifications are constrained to the following subprograms

MISCLN.f
UPP_PHYSICS.f
TTBLEX.f

Code description::
(1) TTBLEX.f (line 120 ~ 240): a new routine TTBLEX1D was added (specifically used for subroutine CALCAPE1D), based on original routine TTBLEX which is CALCAPE for air parcels in 2-D plane.
(2) UPP_PHYSICS.f (line 2580 ~ 3014): adding a new routine CALCAPE1D
(3) MISCLN.f (line 3419 ~ 3749):
adding code to search for the effective layer.
scheme 1: line 3438 ~ 3491. Using CALCAPE to compute CAPE/CIN for all air parcels level by level
scheme 2: line 3492 ~ 3688. Finding air-parcel with max theta-e along the sounding profile/column on each grid box for sanity check first, then compute cape/cin of the all 1-D air parcels along that each profile if the sanity check is passed for this column/profile.
line 3689 ~ 3749: additional output to print out the profiles, positions of EL, etc for code debugging.

line:3934 ~ 3943: Array LLOW and LUPP are used to store the base and top of EL for other related variables (like helicity).

!--- IF USSING EL BASE & TOP COMPUTED BY NEW SCHEME FOR THE RELATED VARIABLES
IF ( EL_SCHEME > 0 ) THEN
!$omp parallel do private(i,j)
DO J=JSTA,JEND
DO I=1,IM
LLOW(I,J) = EL_BASE(I,J)
LUPP(I,J) = EL_TOPS(I,J)
ENDDO
ENDDO
END IF
So if set EL_SCHEME =1 or 2 in the beginning of code, then the EL found by new scheme is used, otherwise the EL of old scheme is used.

Usage:
In MISCLN.f:
line 199: debugprint = .TRUE. (if set to .FALSE., then no additional output. Those outputs are in text format and very huge, better turned off after further testing.)
line 200: EL_SCHEME = 1
any negative number , or 0--> OLD scheme;
1 --> scheme 1;
2 --> scheme 2
Note: if set EL_SCHEME =1 or 2 in the beginning of code, then the EL found by new scheme is used, otherwise the EL of old scheme is used.

EdwardColon-NOAA committed:
Update params_grib2_tbl_new

EdwardColon-NOAA committed:
Update post_avblflds.xml

EdwardColon-NOAA committed:
Merged changes from the EMC develop branch.

EdwardColon-NOAA committed:
This commit includes changes to the computation of the effective layer top and bottom.

EdwardColon-NOAA committed:
This commit includes changes to the computation of the effective layer top and bottom.

WenMeng-NOAA commented on sorc/ncep_post.fd/ALLOCATE_ALL.f:
```diff
@@ -79,7 +79,7 @@ SUBROUTINE ALLOCATE_ALL()
       do l=1,lm
         do j=jsta_2l,jend_2u
-          do i=1,im
+          do i=1,lm
```
@EdwardColon-NOAA May I ask the reason of  changing im into lm?

GangZhao-NOAA replied:
@EdwardColon-NOAA    I have the same questions for reason of this changing im to lm. Ed you might have mentioned and explained this issue before in our internal meeting. I forgot it.

EdwardColon-NOAA replied:
The i loop min.max indices were mistakenly altered when updating the code. My apologies. This problem has been corrected.

WenMeng-NOAA commented on sorc/ncep_post.fd/ALLOCATE_ALL.f:
```diff
@@ -147,7 +147,7 @@ SUBROUTINE ALLOCATE_ALL()
       do l=1,lm
       do j=jsta_2l,jend_2u
-          do i=1,im 
+          do i=1,lm 
```
@EdwardColon-NOAA May I ask the reason of  changing im into lm?

WenMeng-NOAA commented on sorc/ncep_post.fd/ALLOCATE_ALL.f:
```diff
@@ -195,7 +195,7 @@ SUBROUTINE ALLOCATE_ALL()
       do l=1,lm
       do j=jsta_2l,jend_2u
-          do i=1,im
+          do i=1,lm
```
@EdwardColon-NOAA May I ask the reason of  changing im into lm?

WenMeng-NOAA commented on sorc/ncep_post.fd/ALLOCATE_ALL.f:
```diff
@@ -1229,7 +1231,7 @@ SUBROUTINE ALLOCATE_ALL()
 !$omp parallel do private(i,j)
        do j=jsta_2l,jend_2u
-         do i=1,im
+         do i=1,lm
```
@EdwardColon-NOAA May I ask the reason of  changing im into lm?

WenMeng-NOAA commented on sorc/ncep_post.fd/ALLOCATE_ALL.f:
```diff
@@ -1271,7 +1273,7 @@ SUBROUTINE ALLOCATE_ALL()
 !$omp parallel do private(i,j)
       do j=jsta_2l,jend_2u
-        do i=1,im
+        do i=1,lm
```
@EdwardColon-NOAA May I ask the reason of  changing im into lm?

WenMeng-NOAA commented on sorc/ncep_post.fd/ALLOCATE_ALL.f:
```diff
@@ -1285,7 +1287,7 @@ SUBROUTINE ALLOCATE_ALL()
 !$omp parallel do private(i,j)
       do j=jsta_2l,jend_2u
-        do i=1,im
+        do i=1,lm
```
@EdwardColon-NOAA May I ask the reason of  changing im into lm?

GangZhao-NOAA commented on sorc/ncep_post.fd/MISCLN.f:
```diff
@@ -169,11 +173,34 @@ SUBROUTINE MISCLN
 !****************************************************************************
 !     START MISCLN HERE.
 !     
+       debugprint = .TRUE.
```
debugprint was used during the code development to print out more information for special tracing & debugging the scheme 1 and 2 for searching the effective-layer. It is not necessary for routine use. 
I suggest to set "debugprint = .FALSE." as default.

GangZhao-NOAA commented on sorc/ncep_post.fd/MISCLN.f:
```diff
@@ -3422,11 +3457,361 @@ SUBROUTINE MISCLN
+
+        END DO ! L
+!
+      ELSE IF ( EL_SCHEME == 2 ) THEN
```
When I developed the code to search for Effective-Layer, I made two schemes. So the two schemes could compare to each other and verify against each other. After the testing in HWT SFE 2021, it seems that Scheme 1 had been working well and SPC was satisfied with it. 
So to make the code management and maintenance simple and easy, I suggest to set Scheme 1 as the only scheme for Effective-Layer, and remove the code related to Scheme 2 (in this MISCLN.f, and in UPP_PHYSICS.f and in TTBLEX.f). Just my suggestion. These code still can be kept.

WenMeng-NOAA replied:
@EdwardColon-NOAA and @GangZhao-NOAA If scheme 2 is used only by 3DRTMA, I would agree with Gang that keeping scheme 1 only.

GangZhao-NOAA commented on sorc/ncep_post.fd/TTBLEX.f:
```diff
@@ -112,5 +112,132 @@ SUBROUTINE TTBLEX(TREF,TTBL,ITB,JTB,KARR,PMIDL         &
 !&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&
+!
+!-------------------------------------------------------------------------------------
+!
```
As what I comment in MISCLN.f for scheme 2,
 "_After the testing in HWT SFE 2021, it seems that Scheme 1 had been working well and SPC was satisfied with it. So to make the code management and maintenance simple and easy, I suggest to set Scheme 1 as the only scheme for Effective-Layer, and remove the code related to Scheme 2 (in this MISCLN.f, and in UPP_PHYSICS.f and in TTBLEX.f). Just my suggestion. These code still can be kept._"
This subroutine TTBLEX1D is used only for scheme 2 in MISCLN.f for searching for the Effective-Layer. It could be removed for code clean if scheme 2 is not necessary.

GangZhao-NOAA commented on sorc/ncep_post.fd/UPP_PHYSICS.f:
```diff
@@ -1764,6 +1764,440 @@ SUBROUTINE CALCAPE2(ITYPE,DPBND,P1D,T1D,Q1D,L1D,    &
+!
+!
+
+      SUBROUTINE CALCAPE1D(P_A,T_A,Q_A,ZINT_A,LPAR0,PSFC,LMASK,CAPE,CINS,    &
```
Same as what I comment in MISCLN.f for scheme 2 and in TTBLEX.f for subroutine TTBLEX1D,
 "_After the testing in HWT SFE 2021, it seems that Scheme 1 had been working well and SPC was satisfied with it. So to make the code management and maintenance simple and easy, I suggest to set Scheme 1 as the only scheme for Effective-Layer, and remove the code related to Scheme 2 (in this MISCLN.f, and in UPP_PHYSICS.f and in TTBLEX.f). Just my suggestion. These code still can be kept._"
This subroutine CALCAPE1D is used only for scheme 2 in MISCLN.f for searching for the Effective-Layer. It could be removed for code clean if scheme 2 is not necessary.

GangZhao-NOAA reviewed (commented):
I have one question and one comment:

1. one question is about the reason of changing "im to lm" in some code. This is same question as Wen Meng already asked Ed for the reason.
2. One comment is about whether to keep Scheme 2 (searching for Effective-Layer) in MISCLN.f or remove Scheme 2.
As what I comment in MISCLN.f, in TTBLEX.f and in UPP_PHYSCIS.f for scheme 2,
 "_After the testing in HWT SFE 2021, it seems that Scheme 1 had been working well and SPC was satisfied with it. So to make the code management and maintenance simple and easy, I suggest to set Scheme 1 as the only scheme for Effective-Layer, and remove the code related to Scheme 2 (in this MISCLN.f, and in UPP_PHYSICS.f and in TTBLEX.f). Just my suggestion. These code still can be kept if you like._"

The subroutine TTBLEX1D (in TTBLEX.f) , subroutine CALCAPE1D (in UPP_PHYSCIS.f) and code for scheme 2 in (MISCLN.f) are used only for scheme 2 to search for the Effective-Layer. It could be removed for code clean if scheme 2 is not necessary.

If you would like to remove these code related to Scheme 2, I can help to clean it.

Thank you!

- Gang

WenMeng-NOAA commented on sorc/ncep_post.fd/MISCLN.f:
```diff
@@ -1401,7 +1428,9 @@ SUBROUTINE MISCLN
               DO I=1,IM
                 GRID1(I,J)=Z1D(I,J)
-                IF (SUBMODELNAME == 'RTMA') FREEZELVL(I,J)=GRID1(I,J)
+                IF (SUBMODELNAME == 'RTMA' .OR. MODELNAME == 'FV3R') THEN
```
@EdwardColon-NOAA May I ask the reason of adding model name "FV3R"?

EdwardColon-NOAA replied:
This additional conditional  model check (FV3R) was added in order to ensure that the FREEZELVL field would be accessible for the calculation of the SPC Hail parameter which needs to be an output product for the RRFS-based 3D-RTMA workflow.

WenMeng-NOAA commented on sorc/ncep_post.fd/MISCLN.f:
```diff
@@ -2910,7 +2938,8 @@ SUBROUTINE MISCLN
 
-        IF (MODELNAME == 'GFS' .OR. MODELNAME == 'FV3R') THEN
+        IF (MODELNAME == 'GFS' .OR. MODELNAME == 'FV3R' &
+           .OR. MODELNAME == 'FV3R') THEN
```
@EdwardColon-NOAA The model name "FV3R" is duplicated here.

EdwardColon-NOAA replied:
This duplication was resolved.

WenMeng-NOAA commented on sorc/ncep_post.fd/MISCLN.f:
```diff
@@ -3164,7 +3191,9 @@ SUBROUTINE MISCLN
                    IF(T1D(I,J) < spval) THEN
                       GRID1(I,J) = - GRID1(I,J)
-                      IF (SUBMODELNAME == 'RTMA') MLCIN(I,J) = GRID1(I,J)
+                      IF (SUBMODELNAME == 'RTMA' .OR. MODELNAME == 'FV3R') THEN
```
@EdwardColon-NOAA May I ask the reason of adding model name "FV3R"?

EdwardColon-NOAA replied:
This additional conditional model check (FV3R) was added in order to ensure that the MLCIN field would be accessible for the calculation of the effective layer tornado parameter which needs to be an output product for the RRFS-based 3D-RTMA workflow.

WenMeng-NOAA commented on sorc/ncep_post.fd/MISCLN.f:
```diff
@@ -3252,7 +3283,9 @@ SUBROUTINE MISCLN
            CALL CALCAPE(ITYPE,DPBND,P1D,T1D,Q1D,LB2,EGRID1,     &
                         EGRID2,EGRID3,EGRID4,EGRID5)
-           IF (SUBMODELNAME == 'RTMA') MUMIXR(I,J) = Q1D(I,J)
+!           IF (SUBMODELNAME == 'RTMA' .OR. MODELNAME == 'FV3R') THEN
```
May I ask the reason of adding model name "FV3R"?

EdwardColon-NOAA replied:
This additional conditional model check (FV3R) was added in order to ensure that the most unstable mixing ratio field would be accessible as an output product for the RRFS-based 3D-RTMA workflow.

WenMeng-NOAA commented on sorc/ncep_post.fd/MISCLN.f:
```diff
@@ -3261,12 +3294,14 @@ SUBROUTINE MISCLN
                    IF(T1D(I,J) < spval) THEN
                       GRID1(I,J) = EGRID1(I,J)
-                      IF (SUBMODELNAME == 'RTMA') MUCAPE(I,J) = GRID1(I,J)
+                      IF (SUBMODELNAME == 'RTMA' .OR. MODELNAME == 'FV3R') THEN
```
May I ask the reason of adding model name "FV3R"?

EdwardColon-NOAA replied:
This additional conditional model check (FV3R) was added in order to ensure that the field would be accessible for the calculation of the SPC effective layer supercell parameter which needs to be an output product for the RRFS-based 3D-RTMA workflow.

WenMeng-NOAA commented on sorc/ncep_post.fd/MISCLN.f:
```diff
@@ -3298,7 +3333,8 @@ SUBROUTINE MISCLN
                    IF(T1D(I,J) < spval) THEN 
                    GRID1(I,J) = - GRID1(I,J)
-                       IF (SUBMODELNAME == 'RTMA') THEN
+                       IF (SUBMODELNAME == 'RTMA' .OR. &
```
May I ask the reason of adding model name "FV3R"?

GangZhao-NOAA replied:
@WenMeng-NOAA , I guess the model name "FV3R" might be made by the GSL colleagues for their experimental RRFS system. I am not sure about it. Ed (@EdwardColon-NOAA ) should know about this issue. He's working with Guoqing and Ming Hu of GSL on the UPP for RRFS recently.

WenMeng-NOAA replied:
I would like to get confirmation from RRFS developers regarding the changes for RRFS in the PR of 3DRTMA. Please coordinate with them. Thanks!

GangZhao-NOAA replied:
@WenMeng-NOAA , I think Ed (@EdwardColon-NOAA ) could confirm and coordinate with GSL colleagues on this issue. Thanks!

EdwardColon-NOAA replied:
The addition of the RV3R conditional requirement has been added intentionally to allow the production of SPC fields for RRFS-based 3D-RTMA output.

EdwardColon-NOAA committed:
Update ALLOCATE_ALL.f

i loop indices corrected.

EdwardColon-NOAA committed:
Update MISCLN.f

EdwardColon-NOAA committed:
Update MISCLN.f

Removed EL_SCHEME flag and the EL_SCHEME 2 as it was intended as a method of validating the output of EL_SCHEME effective layer top and bottom calculations. The HWT/SFE confirmed that the EL_SCHEME 1 computations were viable making EL_SCHEME 2 superfluous. The debugprint flag was also set to false as it currently not needed in the execution of the MISCLN subprogram, however, it may be switch on at any point it becomes necessary to trace a problem.

EdwardColon-NOAA committed:
Update UPP_PHYSICS.f

Removed the definition of CALCAPE1D which was invoked in MISCLN.f as the CAPE/CIN computational routine used in EL_SCHEME 2. It is now a defunct option.

WenMeng-NOAA commented:
@EdwardColon-NOAA Thanks for updating based on comments. Can you also sync your branch with upstream/develop and solve the conflicts in parm/ppst_avbflds.xml.

EdwardColon-NOAA committed:
Update MISCLN.f

EdwardColon-NOAA committed:
Update MISCLN.f

EdwardColon-NOAA committed:
Fixed a conflict in parm/post_avblflds.xml.

WenMeng-NOAA commented:
@EdwardColon-NOAA Could you  sync your branch with upstream/develop and solve the conflicts in parm/ppst_avbflds.xml so I can conduct the tests?

EdwardColon-NOAA committed:
Merge branch 'feature/3DRTMA8' of https://github.com/EdwardColon-NOAA/EMC_post into feature/3DRTMA8

EdwardColon-NOAA commented:
Hi Wen,

Sorry, I had some issues in setting up personal access tokens needed to
push updates back to the repository after syncing my branch and resolving
the conflict in parm/post_avbflds.xml.

-Edward

On Tue, Aug 17, 2021 at 9:11 AM WenMeng-NOAA ***@***.***>
wrote:

> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA> Could you sync
> your branch with upstream/develop and solve the conflicts in
> parm/ppst_avbflds.xml so I can conduct the tests?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/358#issuecomment-900285732>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQDZJ4K7PDC33HJKHITT5JNYLANCNFSM5BU5DEBQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
-------------------------------------------------
Edward Colón
IMSG at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

WenMeng-NOAA commented:
@EdwardColon-NOAA Now your branch is in a good shape. Thanks!

WenMeng-NOAA commented:
@EdwardColon-NOAA My tests indicate that EFHL in hrrr/rap datasets have changed results like:
163:149027926:EFHL:surface:rpn_corr=0.238913:rpn_rms=37.7674

Do you think what would cause EFHL change?

EdwardColon-NOAA commented:
Hi Wen,

I will need to consult Guoqing since this commit did include FV3R updates
related to RRFS integration.

-Edward

On Tue, Aug 17, 2021 at 11:06 AM WenMeng-NOAA ***@***.***>
wrote:

> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA> My tests indicate
> that EFHL in hrrr/rap datasets have changed results like:
> 163:149027926:EFHL:surface:rpn_corr=0.238913:rpn_rms=37.7674
>
> Do you think what would cause EFHL change?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/358#issuecomment-900380067>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQGOYF3WLNDGLU64HNLT5J3GFANCNFSM5BU5DEBQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>
-- 
-------------------------------------------------
Edward Colón
IMSG at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

EdwardColon-NOAA commented:
Hi Wen,

I determined that the problem stems from the new calculation of effective
layer top and bottom. The EFHL field is impacted as a result and will not
yield a bit-wise identical comparison with the regression test HRRR
reference file. There are a couple of ways around this. One would be to
maintain the old method of calculating the effective layer top and bottom
as an option linked to the selection of the HRRR model input (RAPR) while
the new method would only be invoked if the submodel is specified as RTMA.
The other is just to update the HRRR regression test reference file with
the new calculation of the effective layer top and bottom.  The later would
impact current UPP stakeholders if the EFHL field is a product needed in
their research.

Let me know what you think.


-Edward

On Tue, Aug 17, 2021 at 12:16 PM Edward Colon - NOAA Affiliate <
***@***.***> wrote:

>
> Hi Wen,
>
> I will need to consult Guoqing since this commit did include FV3R updates
> related to RRFS integration.
>
> -Edward
>
> On Tue, Aug 17, 2021 at 11:06 AM WenMeng-NOAA ***@***.***>
> wrote:
>
>> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA> My tests
>> indicate that EFHL in hrrr/rap datasets have changed results like:
>> 163:149027926:EFHL:surface:rpn_corr=0.238913:rpn_rms=37.7674
>>
>> Do you think what would cause EFHL change?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NOAA-EMC/EMC_post/pull/358#issuecomment-900380067>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACFXXQGOYF3WLNDGLU64HNLT5J3GFANCNFSM5BU5DEBQ>
>> .
>> Triage notifications on the go with GitHub Mobile for iOS
>> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
>> or Android
>> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
>> .
>>
> --
> -------------------------------------------------
> Edward Colón
> IMSG at NOAA/NWS/NCEP/EMC
> 5830 University Research Ct.
> Rm #2025
> Riverdale, MD 20737
> Office: (301) 683-3815
> Cell: (301) 213-3566
>


-- 
-------------------------------------------------
Edward Colón
IMSG at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

WenMeng-NOAA commented:
@EdwardColon-NOAA The first way would be the one to get this PR merged quickly. We use separate  algorithms for hrrr/rap and 3Drtma. For the long term solution, you might prepare the evaluation of these two algorithms and coordinate with hrrr/rap developers Geoffrey Manikin and Ming Hu for unification. Please let me know if this proposal is working for you.

EdwardColon-NOAA commented:
Hi Wen,

For the sake of timeliness, I will go ahead and code the algorithm with the
option of using the older method of calculating the effective layer top and
bottom. We will need to revisit the issue with Geoff and Hui-ya later.

On Wed, Aug 18, 2021 at 6:43 PM WenMeng-NOAA ***@***.***>
wrote:

> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA> The first way
> would be the one to get this PR merged quickly. We use separate algorithms
> for hrrr/rap and 3Drtma. For the long term solution, you might prepare the
> evaluation of these two algorithms and coordinate with hrrr/rap developers
> Geoffrey Manikin and Ming Hu for unification. Please let me know if this
> proposal is working for you.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/358#issuecomment-901476633>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQAQ3P7VANB5VPNFANDT5QZSDANCNFSM5BU5DEBQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>
-- 
-------------------------------------------------
Edward Colón
IMSG at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

WenMeng-NOAA commented:
@EdwardColon-NOAA Sounds good to me. Thanks!

EdwardColon-NOAA committed:
Added conditional block that only invokes the effective layer top and bottom correction for RTMA.

WenMeng-NOAA commented:
@EdwardColon-NOAA I would like to know if the scheme 2 mentioned in Gang's comments is removed.

EdwardColon-NOAA commented:
Hi Wen,

I was out of the office this afternoon. Yes, Gang's scheme 2 coding blocks
were removed from this pull request.

Thanks,

Edward

On Fri, Aug 20, 2021 at 2:21 PM WenMeng-NOAA ***@***.***>
wrote:

> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA> I would like to
> know if the scheme 2 mentioned in Gang's comments is removed.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/358#issuecomment-902874284>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQCQHABOA4EAQIYBSADT52MKJANCNFSM5BU5DEBQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
-------------------------------------------------
Edward Colón
IMSG at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

WenMeng-NOAA commented:
@EdwardColon-NOAA Thanks for clarifying. I conducted the regression tests. The hrrr/rap look good. But the 3drtma test failed as:
 get_postfilename,post_fname=WRFTWO.GrbF00npset=           1 num_pset=
           3 iSF_SURFACE_PHYSICS=           3
forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source      
upp.x              00000000014C018D  Unknown               Unknown  Unknown
libpthread-2.17.s  00002AAB9B936630  Unknown               Unknown  Unknown
upp.x              0000000000A1AAD9  miscln_                  3943  MISCLN.f
upp.x              0000000000706547  process_                  116  PROCESS.f
upp.x              000000000065FA79  MAIN__                    865  WRFPOST.f
upp.x              000000000040755E  Unknown               Unknown  Unknown
libc-2.17.so       00002AAB9C220555  __libc_start_main     Unknown  Unknown
upp.x              0000000000407469  Unknown               Unknown  Unknown
 953           42           1           0

Can you take a look at it?

EdwardColon-NOAA committed:
Addressed remaining bug linked to the use of the FV3R first guess.

EdwardColon-NOAA committed:
minor code changes made to ALLOCATE_ALL.f, MISCNL.f and UPP_PHYSICS.f to address segmentation fault occurring during the rtma regression tests

WenMeng-NOAA commented:
@EdwardColon-NOAA You had updated your PR yesterday. Will the current version be working in rtma regression tests, or your tweaking is not done yet?

EdwardColon-NOAA commented:
Hi Wen,

After the minor tweaks, the regression test did run to completion for the
rtma case. They did not yield bitwise identical results which is expected
due to the modified effective layer calculation but I tested it against the
default effective layer calculation and it did yield bitwise results.

-Edward

On Wed, Sep 1, 2021 at 1:44 PM WenMeng-NOAA ***@***.***>
wrote:

> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA> You had updated
> your PR yesterday. Will the current version be working in rtma regression
> tests, or your tweaking is not done yet?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/358#issuecomment-910509785>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQGJ7OCKFHSV63RDEE3T7ZQ6VANCNFSM5BU5DEBQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
-------------------------------------------------
Edward Colón
IMSG at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

WenMeng-NOAA commented:
@EdwardColon-NOAA Thanks for updating and clarifying. I will resume my tests.

WenMeng-NOAA commented:
@EdwardColon-NOAA Now my rtma tests can be run through. Can you sync your branch with upstream/develop and add change logs with your name in the modified fortran code?

EdwardColon-NOAA committed:
Clean up before merging.

EdwardColon-NOAA committed:
Fixed documentation bug.

WenMeng-NOAA commented:
@EdwardColon-NOAA Can you verify the new rtma GRIB2 datasets from my test at  /u/Wen.Meng/ptmp/rtma_2020113000 on Mars? Thanks!

EdwardColon-NOAA commented:
Hi Wen,

I verify the new rtma GRIB2 reference files. I am getting bitwise identical
results with my regression test.

-Edward

On Thu, Sep 2, 2021 at 10:39 AM WenMeng-NOAA ***@***.***>
wrote:

> @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA> Can you verify
> the new rtma GRIB2 datasets from my test at
> /u/Wen.Meng/ptmp/rtma_2020113000 on Mars? Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/358#issuecomment-911758555>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQHCW2O5N34IKJMVQF3T76EBZANCNFSM5BU5DEBQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
-------------------------------------------------
Edward Colón
IMSG at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

WenMeng-NOAA commented:
@EdwardColon-NOAA Thanks! I will recreate the baseline for rtma with this PR merged.

WenMeng-NOAA commented:
The UPP regression tests were conducted on WCOSS-Dell, WCOSS-Cray and Hera. With this PR merged, the baseline for rtma will be recreated.
